server/src/index.ts

```diff
@@ -60,22 +64,61 @@ serve(
       ),
     },
     "/api-token/": {
-      GET: authenticated(verifyAuth0, async (_, uid) => {
-        const token = await redisClient.get(redisKey.userToApiToken(uid));
-        return token ? new Response(token) : Response404();
-      }),
-      POST: authenticated(verifyAuth0, async (_, uid) => {
-        const oldToken = await redisClient.get(redisKey.userToApiToken(uid));
-        if (oldToken) {
-          await redisClient.del(redisKey.apiTokenToUser(oldToken));
-        }
-        const token = crypto.randomUUID();
-        return Promise.all([
-          redisClient.set(redisKey.userToApiToken(uid), token),
-          redisClient.set(redisKey.apiTokenToUser(token), uid),
-        ]).then(() => new Response(token));
-      }),
+      GET: authenticated(
+        verifyAuth0,
+        (_, uid) =>
+          getApiTokens(uid).then((tokens) =>
+            new Response(JSON.stringify(tokens))
+          ),
+      ),
+      POST: authenticated(
+        verifyAuth0,
+        (request, uid) =>
+          request.json().then(
+            (call: CREATE_TOKEN | DELETE_TOKEN) => {
+              if (call.action == "create") {
+                const token = crypto.randomUUID();
+                const tx = redisClient.tx();
+                tx.rpush(redisKey.userToApiTokenSet(uid), token);
+                tx.set(redisKey.apiTokenToUser(token), uid);
+                return tx.flush().then(() =>
+                  new Response(JSON.stringify(token))
+                );
+              } else if (call.action == "delete") {
```

Comment:
can drop `else` after `return`

Reply:
I skip for now, will find a better branching pattern later